### DISCUSSION TITLE
模态框中使用了，source（）来获取表格数据，如何让每次新增，删除，修改完自动刷新当前表格的数据呢？

### DISCUSSION THREAD

**Discussion by kanganxi** [slowlyo/owl-admin#157] [Anything]
目前是有刷新动作，看了返回值，也有返回，但是表格数据不及时回显（只能重新打开模态框才会变化），请问怎么解决呢？代码如下：
` public function index()
    {

        $chapter = $this->service->getTableData();
        $crudId = str_replace('/', '.', request()->path()) . '.crud';

        $crud = amis()->CRUDTable()
            ->headerToolbar([
                $this->createButton('dialog'),
                $this->baseHeaderToolBar()
            ])
            ->id($crudId)
            //->api(  'book_related_chapter?books_id=${id_name}&type=table&page=1&perPage=10')
            ->quickSaveApi($this->getQuickEditPath())
            ->quickSaveItemApi($this->getQuickEditItemPath())
            ->bulkActions([$this->bulkDeleteButton()])
            ->syncLocation(false)
            ->data(['chapter' => $chapter])
            ->source('${chapter}')
            ->headerToolbar([
                $this->createButton('dialog')->data(['id_name' => '${id}'])->onEvent(['actionType'=>'reload']),
              /*  \Slowlyo\OwlAdmin\Renderers\VanillaAction::make()->label('刷新CRUD数据加载请求')
                    ->level('primary')->className('mb-2')
                    ->onEvent(['click' => ['actions' => [['componentId' => 'chapter','actionType' => 'reload',],],],])*/
            ])
            ->columns([
                amis()->TableColumn('id', 'ID')->sortable(),
                amis()->TableColumn('chapter.title', '章节'), 
                amis()->TableColumn('book.title', '课本'),
                amis()->TableColumn('enabled', '是否启用')->quickEdit(['type' => 'switch','mode' => 'inline','saveImmediately' => true,]),
                amis()->TableColumn('created_at', admin_trans('admin.created_at'))->type('datetime')->sortable(),
                //amis()->TableColumn('updated_at', admin_trans('admin.updated_at'))->type('datetime')->sortable(),
                $this->rowActions('dialog')
            ]);
        return $crud;
    }`

**Comment by slowlyo** [comment]
哪的模态框🤨

**Reply by kanganxi** [reply]
抱歉，表述不当，Dialog 对话框空渲染table(列表)，里面有增删改的操作，希望增删改之后，列表能刷新最新的内容

**Reply by slowlyo** [reply]
在 dialog 里面的列表的话, 貌似只能手动调用 crud 组件的动作去刷新, 看一下 [amis 的文档](https://aisuda.bce.baidu.com/amis/zh-CN/components/crud#reload)
